File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l=0,h=nums.size()-1,m;
        while(l<h){
            m=l+(h-l)/2;
            if(nums[m]>nums[m+1]) return nums[(m+1)%nums.size()];
            else if(nums[l]>nums[m]) h=m-1;
            else l=m+1;
        }
        return nums[(l+1)%nums.size()];
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int s=0,e=nums.length-1,mid;
        while(s<e){
            mid =s+(e-s)/2;
            if(nums[mid]>nums[mid+1]) return nums[(mid+1)%nums.length];
            else if(nums[mid]<nums[s]){
                e=mid-1;
            }
            else{
                s = mid+1;
            }
        }
        return nums[(s+1)%nums.length];
    }
}